src/Conversion/ONNXToTOSA/Math/Elementwise.cpp

```diff
@@ -52,14 +75,34 @@ LogicalResult checkBasicTosaRequirementsForBinaryOps(ConversionPatternRewriter &
 
 // Element-wise unary ops lowering to TOSA dialect.
 //===----------------------------------------------------------------------===//
-template <typename ElementwiseUnaryOpONNX, typename ElementwiseUnaryOpTOSA>
+template <typename ElementwiseUnaryOpONNX, typename ElementwiseUnaryOpTOSA,
+    typename InputType, typename OutputType>
 class ONNXElementwiseUnaryOpLoweringToTOSA
     : public OpConversionPattern<ElementwiseUnaryOpONNX> {
 public:
   using OpConversionPattern<ElementwiseUnaryOpONNX>::OpConversionPattern;
   using OpAdaptor = typename ElementwiseUnaryOpONNX::Adaptor;
   LogicalResult matchAndRewrite(ElementwiseUnaryOpONNX op, OpAdaptor adaptor,
       ConversionPatternRewriter &rewriter) const override {
+
+    Value input = *adaptor.getODSOperands(0).begin();
+    auto inputType = input.getType().dyn_cast<TensorType>();
+    Value output = op.getResult();
+    auto outputType = output.getType().dyn_cast<TensorType>();
+
+    if (!inputType || !outputType) {
+      return rewriter.notifyMatchFailure(op, "Tosa only supports TensorTypes");
+    }
+
+    Type inputElementType = inputType.getElementType();
+    Type outputElementType = outputType.getElementType();
+
+    if (failed(InputType::checkType(rewriter, inputElementType, op)))
+      return rewriter.notifyMatchFailure(op, "input type not supported");
```

Comment:
Do you need the additional `notifyMatchFailure` given `checkType` already did so?

Reply:
Good catch, I didn't notice this.